File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function() {
    
};

function createArrayOfNum(maxValue) {
    return [...Array(maxValue).keys()];
  }


function findArmstrongNumbers(arr) {
    let answer = [];
    for (let i = 0; i < arr.length; i++) {
        let numCompare = arr[i];
        let strNum = arr[i].toString();
        let digits = [];
        let num = 0;
        // console.log('numCompare:', numCompare);
        // console.log('strNum:', strNum);

        for (let j = 0; j < strNum.length; j++) {
            digits.push(parseInt(strNum[j]));
        }
        // console.log('digits:', digits)

        for (let j = 0; j < digits.length; j++) {
            let singleDigit = digits[j];
            let expo = digits.length
            num += Math.pow(singleDigit, expo)
        }
        // console.log('num:', num);

        if (num === numCompare) {
            answer.push(num)
        }
    }

    return answer;
}

console.log(findArmstrongNumbers(createArrayOfNum(999)))

File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
import math

def find_armstrong_numbers(numbers):
   answer = [] 
   for num in numbers:
       num_compare = num
       str_num = str(num)
       digits = []
       final_num = 0
       print('num_compare:', num_compare)
       print('str_num:', str_num)
       
       for digit in str_num:
           digits.append(int(digit))
           print('digits:', digits)
           
       for digit in digits:
           final_num += math.pow(digit, len(digits))
           print('final_num:', final_num)
           
       if final_num == num_compare:
           answer.append(int(final_num))

   return answer
     

find_armstrong_numbers()


